crates/sui-core/src/consensus_handler.rs

```diff
@@ -646,238 +740,1639 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
-                        .with_label_values(&[kind])
-                        .observe(parsed.serialized_len as f64);
-                    // UserTransaction exists only when mysticeti_fastpath is enabled in protocol config.
-                    if matches!(
-                        &parsed.transaction.kind,
-                        ConsensusTransactionKind::CertifiedTransaction(_)
-                            | ConsensusTransactionKind::UserTransaction(_)
-                    ) {
-                        self.last_consensus_stats
-                            .stats
-                            .inc_num_user_transactions(author);
+
+                    match &parsed.transaction.kind {
+                        ConsensusTransactionKind::UserTransaction(_)
+                        | ConsensusTransactionKind::CertifiedTransaction(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore capability notifications if !should_accept_consensus_certs()
+                        | ConsensusTransactionKind::CapabilityNotificationV2(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore end of publish messages if !should_accept_consensus_certs()
+                        | ConsensusTransactionKind::EndOfPublish(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore execution time observations if !should_accept_consensus_certs()
+                        // Note: we no longer have to check protocol_config.ignore_execution_time_observations_after_certs_closed()
+                        | ConsensusTransactionKind::ExecutionTimeObservation(_)
+                        // DONE(commit-handler-rewrite): [ssm] ignore jwk votes if !should_accept_consensus_certs()
+                        | ConsensusTransactionKind::NewJWKFetched(_, _, _) => continue,
+                        _ => {},
                     }
-                    if let ConsensusTransactionKind::RandomnessStateUpdate(randomness_round, _) =
-                        &parsed.transaction.kind
-                    {
+                }
+
+                if !initial_reconfig_state.should_accept_tx() {
+                    match &parsed.transaction.kind {
+                        // DONE(commit-handler-rewrite): [ssm] ignore dkg confirmation if !should_accept_tx()
+                        ConsensusTransactionKind::RandomnessDkgConfirmation(_, _)
+                        // DONE(commit-handler-rewrite): [ssm] ignore dkg message if !should_accept_tx()
+                        | ConsensusTransactionKind::RandomnessDkgMessage(_, _) => continue,
+                        _ => {},
+                    }
+                }
+
+                match &parsed.transaction.kind {
+                    ConsensusTransactionKind::UserTransaction(_) => {
+                        // DONE(commit-handler-rewrite): ignore mfp user transaction if mfp is disabled
+                        if !self.epoch_store.protocol_config().mysticeti_fastpath() {
```

Comment:
The styling of all the checks are not very consistent.
The first part of checks follow the style of "if some condition { match { ...} }", but then later we are doing "match { if some condition }".
I find it hard to read to be honest.

Reply:
cleaned it up to be more consistent